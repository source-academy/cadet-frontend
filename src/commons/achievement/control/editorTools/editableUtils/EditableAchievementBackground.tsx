import { Button, Dialog, EditableText } from '@blueprintjs/core';
import { useState } from 'react';

type EditableAchievementBackgroundProps = {
  cardTileUrl: string;
  setcardTileUrl: any;
};

function EditableAchievementBackground(props: EditableAchievementBackgroundProps) {
  const { cardTileUrl, setcardTileUrl } = props;

  const [isDialogOpen, setDialogOpen] = useState<boolean>(false);

  return (
    <div>
      <div>
        <Button text={'Edit Background'} onClick={() => setDialogOpen(!isDialogOpen)} />
      </div>
      <Dialog
        onClose={() => setDialogOpen(!isDialogOpen)}
        isOpen={isDialogOpen}
        title={'Edit Background'}
        usePortal={false}
      >
        <EditableText
          placeholder={`Enter your image URL here`}
          value={cardTileUrl}
          onChange={setcardTileUrl}
          multiline={true}
        />
      </Dialog>
    </div>
  );
}

export default EditableAchievementBackground;
