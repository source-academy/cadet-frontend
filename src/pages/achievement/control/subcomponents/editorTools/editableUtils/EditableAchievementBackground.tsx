import { Button, Dialog, EditableText } from '@blueprintjs/core';
import React, { useState } from 'react';

type EditableAchievementBackgroundProps = {
  backgroundImageUrl: string;
  setBackgroundImageUrl: any;
};

function EditableAchievementBackground(props: EditableAchievementBackgroundProps) {
  const { backgroundImageUrl, setBackgroundImageUrl } = props;

  const [isDialogOpen, setDialogOpen] = useState<boolean>(false);

  return (
    <div>
      <div>
        <Button text={'Edit Background'} onClick={() => setDialogOpen(!isDialogOpen)} />
      </div>
      <Dialog
        onClose={() => setDialogOpen(!isDialogOpen)}
        isOpen={isDialogOpen}
        title={'Edit Background'}
        usePortal={false}
      >
        <EditableText
          placeholder={`Enter your image URL here`}
          value={backgroundImageUrl}
          onChange={setBackgroundImageUrl}
          multiline={true}
        />
      </Dialog>
    </div>
  );
}

export default EditableAchievementBackground;
