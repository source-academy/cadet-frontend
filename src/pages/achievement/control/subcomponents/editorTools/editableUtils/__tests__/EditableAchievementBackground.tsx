import { mount } from 'enzyme';
import React from 'react';

import EditableAchievementBackground from '../EditableAchievementBackground';

const mockProps = {
  backgroundImageUrl: '',
  setBackgroundImageUrl: () => {}
};

test('EditableAchievementBackground component renders correctly', () => {
  const goal = <EditableAchievementBackground {...mockProps} />;
  const tree = mount(goal);
  expect(tree.debug()).toMatchSnapshot();
});
