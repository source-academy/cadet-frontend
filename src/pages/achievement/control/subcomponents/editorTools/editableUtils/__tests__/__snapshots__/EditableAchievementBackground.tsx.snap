// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableAchievementBackground component renders correctly 1`] = `
"<EditableAchievementBackground backgroundImageUrl=\\"\\" setBackgroundImageUrl={[Function: setBackgroundImageUrl]}>
  <div>
    <div>
      <Blueprint3.Button text=\\"Edit Background\\" onClick={[Function: onClick]}>
        <button type=\\"button\\" onClick={[Function: onClick]} className=\\"bp3-button\\" disabled={[undefined]} onKeyDown={[Function (anonymous)]} onKeyUp={[Function (anonymous)]} tabIndex={[undefined]}>
          <Blueprint3.Icon icon={[undefined]} />
          <span className=\\"bp3-button-text\\">
            Edit Background
          </span>
          <Blueprint3.Icon icon={[undefined]} />
        </button>
      </Blueprint3.Button>
    </div>
    <Blueprint3.Dialog onClose={[Function: onClose]} isOpen={false} title=\\"Edit Background\\" usePortal={false} canOutsideClickClose={true}>
      <Blueprint3.Overlay onClose={[Function: onClose]} isOpen={false} title=\\"Edit Background\\" usePortal={false} canOutsideClickClose={true} className=\\"bp3-overlay-scroll-container\\" hasBackdrop={true} autoFocus={true} backdropProps={{...}} canEscapeKeyClose={true} enforceFocus={true} lazy={true} transitionDuration={300} transitionName=\\"bp3-overlay\\" />
    </Blueprint3.Dialog>
  </div>
</EditableAchievementBackground>"
`;
